components/constellation/constellation.rs

```diff
@@ -1823,7 +1836,14 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
         if webdriver_reset {
             self.webdriver.load_channel = None;
         }
-        self.compositor_proxy.send(ToCompositorMsg::LoadComplete);
+        let pipeline_is_top_level_pipeline = self.browsing_contexts
+            .get(&BrowsingContextId::from(top_level_browsing_context_id))
+            .map(|ctx| ctx.pipeline_id == pipeline_id)
+            .unwrap_or(false);
+        if pipeline_is_top_level_pipeline {
+            // Notify embedder top level document finished loading.
+            self.compositor_proxy.send(ToCompositorMsg::LoadComplete(top_level_browsing_context_id));
+        }
```

Comment:
The difference IIRC is how mozbrowser iframes are treated. Once mozbrowser iframes are removed we can just check parent_info, but for the moment it does make a difference which one we do.